pkg/handlers/aws/mutation/region/inject_test.go

```diff
@@ -25,11 +28,39 @@ var _ = Describe("Generate AWS Region patches", func() {
 	patchGenerator := func() mutation.GeneratePatches {
 		return mutation.NewMetaGeneratePatchesHandler("", NewPatch()).(mutation.GeneratePatches)
 	}
-	regiontests.TestGeneratePatches(
-		GinkgoT(),
-		patchGenerator,
-		clusterconfig.MetaVariableName,
-		v1alpha1.AWSVariableName,
-		VariableName,
-	)
+
+	testDefs := []capitest.PatchTestDef{
+		{
+			Name: "unset variable",
+		},
+		{
+			Name: "region set",
+			Vars: []runtimehooksv1.Variable{
+				capitest.VariableWithValue(
+					clusterconfig.MetaVariableName,
+					"a-specific-region",
+					v1alpha1.AWSVariableName,
+					VariableName,
+				),
+			},
+			RequestItem: request.NewAWSClusterTemplateRequestItem("1234"),
+			ExpectedPatchMatchers: []capitest.JSONPatchMatcher{{
+				Operation:    "add",
+				Path:         "/spec/template/spec/region",
+				ValueMatcher: gomega.Equal("a-specific-region"),
+			}},
+		},
+	}
+
+	// create test node for each case
+	for testIdx := range testDefs {
```

Comment:
So this is now needed because `AssertGeneratePatches` doesn't build the table? This needs to be repeated for every test, why is it being moved out of `ValidateGeneratePatches`?

Reply:
My thinking behind it was to have more control over test execution and keep most of the test execution cycle. for example, If I want to do other additional validation for a httpProxy variable that does not apply to other variables, I can easily add it here.